src/vs/workbench/parts/terminal/electron-browser/terminalInstance.ts

```diff
@@ -135,6 +139,18 @@ export class TerminalInstance implements ITerminalInstance {
 		this._createProcess(this._shellLaunchConfig);
 		this._createXterm();
 
+		this._pidStack = [];
+		this._checkWindowShell = new Emitter<string>();
+		debounceEvent(this._checkWindowShell.event, (l, e) => e, 100, true)
```

Comment:
This should only be called on Windows, otherwise we're still doing a lot of work on enter for nothing on macOS/Linux.